README.md

```diff
@@ -63,9 +63,85 @@ Observer pattern. (2022, November 13). Wikipedia. https://en.wikipedia.org/wiki/
 #### References
 
 
+## Functional Patterns vs. OOP Design Patterns in JavaScript
 
+### Introduction: 
 
+In traditional Object-Oriented Programming (OOP), design patterns are widely used to solve common problems by providing structured approaches to design and development. However, with the rise of Functional Programming (FP), there's a shift towards different paradigms and patterns. In this Section, we'll explore the contrast between OOP design patterns and functional patterns in JavaScript.
 
+### OOP Design Patterns
+OOP design patterns like Singleton, Factory, Observer, and others are well-established concepts in JavaScript development. They focus on class hierarchies, encapsulation, and inheritance to achieve modularity and extensibility.
+
+For example, the Singleton pattern ensures that a class has only one instance and provides a global point of access to it. In JavaScript:
+
+```
+This example is taken from this repository's Singleton JS code.
+
+/**
+ * Singleton class
+ *
+ * @class
+ * @returns {AlertDialog} The AlertDialog class instance.
+ */
+
+class Singleton {
+	constructor() {
+		if (Singleton.instance) {
+			return Singleton.instance;
+		}
+		this.data = "This is the singleton instance.";
+
+		// Save the instance as a static property
+		Singleton.instance = this;
+
+		return this;
+	}
+
+	getData() {
+		return this.data;
+	}
+}
+
+const instance1 = new Singleton();
+console.log(instance1.getData());
+// Output: This is the singleton instance.
+
+const instance2 = new Singleton();
+console.log(instance2.getData());
+// Output: This is the singleton instance.
+
+console.log(instance1 === instance2); // Output: true
+```
+
+### Functional Patterns
+Functional programming emphasizes immutable data and pure functions without side effects. While traditional OOP design patterns may not directly translate, functional programming offers its own set of patterns to address similar concerns.
+
+One common pattern in FP is the Higher-Order Function pattern. Higher-order functions take other functions as arguments or return them, enabling composition and abstraction.
+
+```
+// Example of map, a higher-order function
+const numbers = [1, 2, 3, 4, 5];
+const doubled = numbers.map(x => x * 2);
+console.log(doubled); // [2, 4, 6, 8, 10]
+```

```

Comment:
Great example. I think it could be improved with just a little more information for those who might be unfamiliar with the term "higher order function" and with JS arrow functions. Can you add a bit more text that serves to explain that `map` is a higher-order function, and that explains that the arrow function is an actual JS function?